.travis.yml

```diff
@@ -88,3 +99,43 @@ matrix:
         - ./tests/scripts/wait-for-alfresco-start.sh "http://localhost:8082/alfresco"
       script:
         - travis_wait 30 mvn install -q -f tests/tas-integration/pom.xml -Pall-tas-tests -Denvironment=default -DrunBugs=false
+    - stage: Engineering Release
+      name: "Maven release"
+      if: commit_message ~= /\[trigger release\]/ AND fork = false AND branch =~ ^(master|develop)$ AND type != pull_request 
+      install: travis_retry travis_wait 40 mvn clean install -DskipTests=true "-Dversion.edition=${VERSION_EDITION}" -Dmaven.javadoc.skip=true -B -V
+      # Fail the job if there is a docker image tag that matches the RELEASE_VERSION (or project version in the POM if not set)
+      before_script: ./scripts/travis/verify_release_tag.sh 
+      script: travis_wait 40 ./scripts/travis/maven_release.sh ${RELEASE_VERSION} ${DEVELOPMENT_VERSION}
+      # Copy alfresco.war and the distribution zip to deploy_dir
+      before_deploy: 
+         # Move the final artifacts to a single folder (deploy_dir) to be copied to S3
+        - mkdir -p deploy_dir
+        - cp war/target/alfresco.war deploy_dir
+        - cp distribution/target/*-distribution*.zip deploy_dir
+      deploy:
+      - provider: s3
+        access_key_id: ${AWS_STAGING_ACCESS_KEY}
+        secret_access_key: ${AWS_STAGING_SECRET_KEY}
+        bucket: "alfresco-artefacts-staging"
+        region: "eu-west-1"
+        skip_cleanup: true
+        acl: private
+        local_dir: deploy_dir
+        upload_dir: "alfresco-content-services-community/release/${TRAVIS_BRANCH}/${TRAVIS_BUILD_NUMBER}"
+        on: 
+          all_branches: true
+          condition: $TRAVIS_BRANCH =~ ^(master|develop)$
+    - stage: "Company Release"
```

Comment:
Is "Company Release" and [trigger company release] the correct names for this stage? It does not actually release anything it just publishing artifacts to the release bucket. We could simply call the previous stage "release" and this stage "publish". Otherwise someone could "[trigger company release]", but since this is not an accurate name of that this stage does nothing will actually be released (or published). I know that this is how it was on bamboo, but maybe can be improved